lib/grpcServer/handlers/createGrpcErrorFromDriveResponse.js

```diff
@@ -0,0 +1,72 @@
+const cbor = require('cbor');
+const {
+  server: {
+    error: {
+      InternalGrpcError,
+      InvalidArgumentGrpcError,
+      DeadlineExceededGrpcError,
+      ResourceExhaustedGrpcError,
+      NotFoundGrpcError,
+      FailedPreconditionGrpcError,
+      UnavailableGrpcError,
+      GrpcError,
+    },
+  },
+} = require('@dashevo/grpc-common');
+const GrpcErrorCodes = require('@dashevo/grpc-common/lib/server/error/GrpcErrorCodes');
+const createConsensusError = require('@dashevo/dpp/lib/errors/consensus/createConsensusError');
+const AlreadyExistsGrpcError = require('@dashevo/grpc-common/lib/server/error/AlreadyExistsGrpcError');
+const VersionMismatchGrpcError = require('@dashevo/grpc-common/lib/server/error/VersionMismatchGrpcError');
+
+/**
+ * @typedef createGrpcErrorFromDriveResponse
+ * @param {number} code
+ * @param {string} info
+ * @return {GrpcError}
+ */
+function createGrpcErrorFromDriveResponse(code, info) {
+  const decodedInfo = cbor.decode(Buffer.from(info, 'base64'));
+
+  // eslint-disable-next-line default-case
+  switch (code) {
+    case GrpcErrorCodes.INVALID_ARGUMENT:
+      return new InvalidArgumentGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.DEADLINE_EXCEEDED:
+      return new DeadlineExceededGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.NOT_FOUND:
+      return new NotFoundGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.ALREADY_EXISTS:
+      return new AlreadyExistsGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.RESOURCE_EXHAUSTED:
+      return new ResourceExhaustedGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.FAILED_PRECONDITION:
+      return new FailedPreconditionGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.INTERNAL:
+      return new InternalGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.UNAVAILABLE:
+      return new UnavailableGrpcError(decodedInfo.message, decodedInfo.metadata);
+    case GrpcErrorCodes.VERSION_MISMATCH:
```

Comment:
Do we still have it?

Reply:
We have it in grpc-common error codes

Reply:
Yes but this is not a standard grpc error and we don’t use it anymore. It was for the previous implementation of DAPI versioning.